From b06ac64e73a9110821a19b84eb6cd3b1dba2e6e8 Mon Sep 17 00:00:00 2001
From: Haiying Wang <Haiying.wang@freescale.com>
Date: Sat, 8 Aug 2015 07:26:32 -0400
Subject: [PATCH 099/508] vfio/fsl_mc: use caccheable non-shareable memory for
 cena

QBMan CENA portal memory needs to be cacheable and non-shareable.

Signed-off-by: Haiying Wang <Haiying.wang@freescale.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index 18f263c..20ae567 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -279,7 +279,7 @@ static int vfio_fsl_mc_mmap(void *device_data, struct vm_area_struct *vma)
 
 #define QBMAN_SWP_CENA_BASE 0x818000000ULL
 	if ((addr & 0xFFF000000) == QBMAN_SWP_CENA_BASE)
-		vma->vm_page_prot = pgprot_cached(vma->vm_page_prot);
+		vma->vm_page_prot = pgprot_cached_ns(vma->vm_page_prot);
 	else
 		vma->vm_page_prot = pgprot_noncached(vma->vm_page_prot);
 
-- 
1.7.5.4

